kong/clustering/services/sync/hooks.lua

```diff
@@ -0,0 +1,191 @@
+local _M = {}
+local _MT = { __index = _M, }
+
+
+local hooks = require("kong.hooks")
+local EMPTY = require("kong.tools.table").EMPTY
+
+
+local ipairs = ipairs
+local ngx_null = ngx.null
+local ngx_log = ngx.log
+local ngx_ERR = ngx.ERR
+local ngx_DEBUG = ngx.DEBUG
+
+
+local DEFAULT_PAGE_SIZE = 512
+
+
+function _M.new(strategy)
+  local self = {
+    strategy = strategy,
+  }
+
+  return setmetatable(self, _MT)
+end
+
+
+local function get_all_nodes_with_sync_cap()
+  local res, err = kong.db.clustering_data_planes:page(DEFAULT_PAGE_SIZE)
```

Comment:
Should we not use `:each()` here instead of simply fetching one page? It seems that this function will only ever return `DEFAULT_PAGE_SIZE` entries even if there are more in the database.